### PR TITLE
Update circleci base image with TF version bump (#3355)

### PR DESCRIPTION
**Pull request by yeounoh**
* Pin tf to 98fdeb774845c0396ed4d35b0ac4665b64ee2919
* Use absl::Hash instead of ir::Literal::Hash
* Requires libtpu-nightly 0.1.dev20220128
* Additional PyTorch requirements for build
* Consolidate conda envs to base
* Install debugging package and bump up the build image tag

### CODE REVIEW

**Review by JackCaoG — APPROVED** [review]
Thanks. Please wait for pytorch side pr to be merged first and our CI pass before merging

**Comment by yeounoh** [comment]
The upstream change is picked up by 1.11 release branch. Merging this PR subsequently.

**Pull request merged**
